tests/pom.xml

```diff
@@ -38,19 +39,46 @@
     <build>
         <plugins>
             <plugin>
-                <groupId>org.jacoco</groupId>
-                <artifactId>jacoco-maven-plugin</artifactId>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
                 <executions>
                     <execution>
-                        <id>report</id>
+                        <id>default-jar</id>
                         <goals>
-                            <goal>report-aggregate</goal>
+                            <goal>jar</goal>
                         </goals>
-                        <phase>verify</phase>
+                        <configuration>
+                            <skipIfEmpty>true</skipIfEmpty>
+                        </configuration>
                     </execution>
                 </executions>
             </plugin>
         </plugins>
     </build>
 
+    <profiles>
+        <profile>
+            <id>coverage</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.jacoco</groupId>
+                        <artifactId>jacoco-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>report-aggregate</id>
+                                <phase>verify</phase>
+                                <goals>
+                                    <goal>report-aggregate</goal>
+                                </goals>
+                            </execution>
+                        </executions>
```

Comment:
It's my hope to add either SonarCloud or Codecov to the CI/CD pipeline soon, either of which will use this aggregated JaCoCo test coverage report

Reply:
See my comment above. Would appreciate some peer review of the CI-based analysis setup